mongodb-datastore/restapi/configure_mongodb_datastore.go

```diff
@@ -87,20 +87,18 @@ func configureAPI(api *operations.MongodbDatastoreAPI) http.Handler {
 	return setupGlobalMiddleware(api.Serve(setupMiddlewares))
 }
 
-func startControlPlane(ctx context.Context, api *operations.MongodbDatastoreAPI, eventRequestHandler controlplane.Integration) error {
+func startControlPlane(ctx context.Context, api *operations.MongodbDatastoreAPI, eventRequestHandler controlplane.Integration, log *log.Logger) error {
 	// 1. create a subscription source
-	natsConnector, err := nats.ConnectFromEnv()
-	if err != nil {
-		return err
-	}
+	natsConnector := nats.NewFromEnv()
+	nats.WithLogger(log)
 	eventSource := eventsource.New(natsConnector)
-
+	eventsource.WithLogger(log)
 	// 2. Create a fixed event subscription with no uniform
 	subSource := subscriptionsource.NewFixedSubscriptionSource(subscriptionsource.WithFixedSubscriptions(keptnapi.EventSubscription{Event: "sh.keptn.event.>"}))
-
+	subscriptionsource.WithLogger(log)
 	// 3. Create control plane
 	controlPlane := controlplane.New(subSource, eventSource, nil)
-
+	controlplane.WithLogger(log)
```

Comment:
actually these lines do not set the logger.
You need to pass it to the New constructor function, e.g.:
`eventsource.New(connector,eventsource.WithLogger(logger))`